File: voiceping-sdk/src/main/java/com/smartwalkie/voicepingsdk/listeners/OutgoingAudioListener.java
package com.smartwalkie.voicepingsdk.listeners;

import com.smartwalkie.voicepingsdk.exceptions.PingException;
import com.smartwalkie.voicepingsdk.models.Message;

public interface OutgoingAudioListener {

    void onMessageReceived(Message message);

    void onSendMessageFailed(byte[] data, PingException e);

    void onConnectionFailure();
}

File: voiceping-sdk/src/main/java/com/smartwalkie/voicepingsdk/listener/OutgoingAudioListener.kt
package com.smartwalkie.voicepingsdk.listener

import com.smartwalkie.voicepingsdk.exception.VoicePingException
import com.smartwalkie.voicepingsdk.model.Message

interface OutgoingAudioListener {

    fun onMessageReceived(message: Message)

    fun onSendMessageFailed(data: ByteArray, e: VoicePingException)

    fun onConnectionFailure(e: VoicePingException)
}